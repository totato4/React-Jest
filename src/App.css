.App {
  text-align: center;
  min-height: 100vh;
  background-color: rgb(166, 140, 140);
  padding: 100px 0px;
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  gap: 50px 0px;
  color: white;
  align-items: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.list-item {
  list-style-type: none;
}
