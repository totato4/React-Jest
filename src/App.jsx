import "./App.css";
import React from "react";
import List from "./components/List/List";
import { Search } from "./components/Search";
const data = [
  "HTML",
  "CSS",
  "JavaScript",
  "TypeScript",
  "React",
  "Vue",
  "Angular",
  "NodeJS",
];

function App() {
  const [search, setSearch] = React.useState("");
  const [items, setItems] = React.useState(data);
  React.useEffect(() => {
    setItems(
      data.filter((el) => el.toLowerCase().includes(search.toLowerCase()))
    );
  }, [search]);

  return (
    <div className="App">
      <div>
        Тестирование реакт приложений <br />
        Пет проект
      </div>
      <Search value={search} onChange={(e) => setSearch(e.target.value)}>
        Find course:
      </Search>
      <List items={items} />
    </div>
  );
}

export default App;
