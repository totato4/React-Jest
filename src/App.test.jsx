import { render, screen } from "@testing-library/react";
import App from "./App";
import userEvent from "@testing-library/user-event";

describe("App component", () => {
  it("App renders", () => {
    render(<App />);
    expect(screen.getByRole("list")).toBeInTheDocument();
    expect(screen.getByText(/Пет проект/i)).toBeInTheDocument();
  });
  it("typing in Searchbox works", () => {
    render(<App />);
    expect(screen.queryByDisplayValue(/React/i)).toBeNull();
    userEvent.type(screen.getByRole("textbox"), "React");
    expect(screen.queryByDisplayValue(/React/i)).toBeInTheDocument();
  });
  it("search filter is working", () => {
    render(<App />);
    expect(screen.getByText(/Vue/i)).toBeInTheDocument();
    expect(screen.getByText(/Javascript/i)).toBeInTheDocument();
    userEvent.type(screen.getByRole("textbox"), "script"),
      expect(screen.queryByText(/Vue/i)).toBeNull();
    expect(screen.getByText(/Javascript/i)).toBeInTheDocument();
  });
  it("app snapshot", () => {
    const app = render(<App />);
    expect(app).toMatchSnapshot();
  });
});
