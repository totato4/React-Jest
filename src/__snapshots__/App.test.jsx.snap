// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component app snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div>
          Тестирование реакт приложений 
          <br />
          Пет проект
        </div>
        <label
          class="label"
        >
          Find course:
          <input
            class="input"
            placeholder="search..."
            type="text"
            value=""
          />
        </label>
        <ul>
          <li
            class="list-item"
          >
            HTML
          </li>
          <li
            class="list-item"
          >
            CSS
          </li>
          <li
            class="list-item"
          >
            JavaScript
          </li>
          <li
            class="list-item"
          >
            TypeScript
          </li>
          <li
            class="list-item"
          >
            React
          </li>
          <li
            class="list-item"
          >
            Vue
          </li>
          <li
            class="list-item"
          >
            Angular
          </li>
          <li
            class="list-item"
          >
            NodeJS
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div>
        Тестирование реакт приложений 
        <br />
        Пет проект
      </div>
      <label
        class="label"
      >
        Find course:
        <input
          class="input"
          placeholder="search..."
          type="text"
          value=""
        />
      </label>
      <ul>
        <li
          class="list-item"
        >
          HTML
        </li>
        <li
          class="list-item"
        >
          CSS
        </li>
        <li
          class="list-item"
        >
          JavaScript
        </li>
        <li
          class="list-item"
        >
          TypeScript
        </li>
        <li
          class="list-item"
        >
          React
        </li>
        <li
          class="list-item"
        >
          Vue
        </li>
        <li
          class="list-item"
        >
          Angular
        </li>
        <li
          class="list-item"
        >
          NodeJS
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
